app/controllers/CalculationController.scala

```diff
@@ -83,7 +97,54 @@ class CalculationController @Inject()(viewLPP: CalculationLPPView,
             val daysSince31 = ChronoUnit.DAYS.between(penalty.get.period.dueDate.plusDays(31), LocalDateTime.now())
             val isEstimate = penalty.get.status.equals(PointStatusEnum.Estimated)
-            Ok(viewAdd(daysSince31, isEstimate, additionalPenaltyRate, startDateOfPeriod, endDateOfPeriod, penaltyAmount, amountReceived,amountLeftToPay))
+            Ok(viewAdd(daysSince31, isEstimate, additionalPenaltyRate, startDateOfPeriod, endDateOfPeriod, penaltyAmount, amountReceived, amountLeftToPay))
+          }
+        }
+      }
+    }
+  }
+
+  def getPenaltyDetailsFromNewAPI(penaltyId: String, isAdditional: Boolean)(implicit request: User[_]): Future[Result] = {
+    logger.debug(s"[CalculationController][getPenaltyDetailsFromNewAPI] - Bar Foo!")
+    penaltiesService.getPenaltyDetailsFromEnrolmentKey(EnrolmentKeys.constructMTDVATEnrolmentKey(request.vrn)).map {
+      payload => {
+        val penalty: Option[LPPDetails] = payload.latePaymentPenalty.flatMap(_.details.find(_.principalChargeReference == penaltyId))
+        if (penalty.isEmpty) {
+          logger.error("[CalculationController][onPageLoad] - Tried to render calculation page with new model but could not find penalty specified.")
+          errorHandler.showInternalServerError
+        } else {
+          val startDateOfPeriod: String = calculationPageHelper.getDateAsDayMonthYear(penalty.get.principalChargeBillingFrom)
+          val endDateOfPeriod: String = calculationPageHelper.getDateAsDayMonthYear(penalty.get.principalChargeBillingTo)
+          val amountReceived = CurrencyFormatter.parseBigDecimalToFriendlyValue(penalty.get.penaltyAmountPaid.get)
+          val isPenaltyEstimate = penalty.get.penaltyStatus.equals(LPPPenaltyStatusEnum.Accruing)
+          val amountLeftToPay = CurrencyFormatter.parseBigDecimalToFriendlyValue(penalty.get.penaltyAmountOutstanding.get)
+          val penaltyAmount = penalty.get.penaltyAmountOutstanding.get + penalty.get.penaltyAmountPaid.get
+          val parsedPenaltyAmount = CurrencyFormatter.parseBigDecimalToFriendlyValue(penaltyAmount)
+          logger.debug(s"[CalculationController][onPageLoad] - found penalty: ${penalty.get}")
+          if (!isAdditional) {
+            val penaltyEstimateDate = penalty.get.principalChargeDueDate.plusDays(30)
+            val calculationRow = calculationPageHelper.getCalculationRowForLPPForNewAPI(penalty.get)
+            calculationRow.fold({
+              //TODO: log a PD
+              logger.error("[CalculationController][onPageLoad] - " +
+                "Calculation row returned None - this could be because the user did not have a defined amount after 15 and/or 30 days of due date")
+              errorHandler.showInternalServerError
+            })(
+              rowSeq => {
+                val isTwoCalculations: Boolean = rowSeq.size == 2
+                val warningPenaltyAmount = CurrencyFormatter.parseBigDecimalToFriendlyValue(penalty.get.penaltyAmountOutstanding.get * 2)
```

Comment:
Raised with design and ticket created, if the user were to part pay their VAT or part pay their penalty then this data would be misaligned to the true value of the VAT owed. Will merge in with this consideration